### PR TITLE
Fix man and commandline docs

### PR DESCRIPTION
**Pull request by runcom**
This patch fixes:
- missing help option in `docs/reference/commandline/*.md` (some files have it, the other I fixed didn't)
- missing `[OPTIONS]` in Usage description
- missing options
- formatting
- start/stop idempotence

Signed-off-by: Antonio Murdaca amurdaca@redhat.com

### CODE REVIEW

**Comment by runcom** [comment]
ping @thaJeztah

File: docs/reference/commandline/build.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ weight=1
       --cpuset-cpus=""         CPUs in which to allow execution, e.g. `0-3`, `0,1`
       --cgroup-parent=""       Optional parent cgroup for the container
       --ulimit=[]              Ulimit options
+      --help=false             Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/build.md]: can you add this in alphabetical order?

File: docs/reference/commandline/commit.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ weight=1
       -c, --change=[]     Apply specified Dockerfile instructions while committing the image
       -m, --message=""    Commit message
       -p, --pause=true    Pause container during commit
+      --help=false        Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/commit.md]: same here (alphabetical)

File: docs/reference/commandline/cp.md [hunk 3]
```diff
@@ -11,12 +11,12 @@ weight=1
...
-Copy files/folders between a container and the local filesystem.
+    Usage: docker cp [OPTIONS] CONTAINER:PATH LOCALPATH|-
+           docker cp [OPTIONS] LOCALPATH|- CONTAINER:PATH
 
-    Usage:  docker cp [options] CONTAINER:PATH LOCALPATH|-
-            docker cp [options] LOCALPATH|- CONTAINER:PATH
+    Copy files/folders between a container and the local filesystem
 
-    --help  Print usage statement
+      --help=false        Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/cp.md]: have you checked the generated output here? Changing indentation in Markdown can be tricky (not sure if it's troublesome in the man-pages here though)

File: docs/reference/commandline/events.md [hunk 4]
```diff
@@ -18,6 +18,7 @@ weight=1
       -f, --filter=[]    Filter output based on conditions provided
       --since=""         Show all events created since timestamp
       --until=""         Stream events until this timestamp
+      --help=false       Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/events.md]: (alphabetical)

File: docs/reference/commandline/exec.md [hunk 5]
```diff
@@ -20,6 +20,7 @@ weight=1
       --privileged=false         Give extended Linux capabilities to the command
       -t, --tty=false            Allocate a pseudo-TTY
       -u, --user=                Username or UID (format: <name|uid>[:<group|gid>])
+      --help=false               Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/exec.md]: same here (alphabetical)

File: docs/reference/commandline/export.md [hunk 6]
```diff
@@ -12,19 +12,17 @@ weight=1
...
-    
-      Export the contents of a filesystem to a tar archive (streamed to STDOUT by default).
 
-      -o, --output=""    Write to a file, instead of STDOUT
-
-      Produces a tarred repository to the standard output stream.
+    Export the contents of a container's filesystem as a tar archive
 
+      -o, --output=""    Write to a file, instead of STDOUT
+      --help=false       Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/export.md]: same here (alphabetical)

File: docs/reference/commandline/history.md [hunk 7]
```diff
@@ -18,6 +18,7 @@ weight=1
       -H, --human=true     Print sizes and dates in human readable format
       --no-trunc=false     Don't truncate output
       -q, --quiet=false    Only show numeric IDs
+      --help=false         Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/history.md]: and here

File: docs/reference/commandline/import.md [hunk 8]
```diff
@@ -19,6 +19,7 @@ weight=1
 
       -c, --change=[]     Apply specified Dockerfile instructions while importing the image
       -m, --message=      Set commit message for imported image
+      --help=false        Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/import.md]: and here :innocent:

File: docs/reference/commandline/inspect.md [hunk 9]
```diff
@@ -15,10 +15,10 @@ weight=1
 
     Return low-level information on a container or image
 
-      -f, --format=""    Format the output using the given go template
-
-     --type=container|image  Return JSON for specified type, permissible
```
Comment (thaJeztah) [inline docs/reference/commandline/inspect.md]: this trailing whitespace can be intentional to force a line-break, please check

File: docs/reference/commandline/kill.md [hunk 10]
```diff
@@ -16,6 +16,7 @@ weight=1
     Kill a running container using SIGKILL or a specified signal
 
       -s, --signal="KILL"    Signal to send to the container
+      --help=false           Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/kill.md]: (alphabetical)

File: docs/reference/commandline/logs.md [hunk 11]
```diff
@@ -19,6 +19,7 @@ weight=1
       --since=""                Show logs since timestamp
       -t, --timestamps=false    Show timestamps
       --tail="all"              Number of lines to show from the end of the logs
+      --help=false              Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/logs.md]: alpha ... well, you know :smile:

File: docs/reference/commandline/ps.md [hunk 12]
```diff
@@ -24,7 +24,8 @@ weight=1
       -q, --quiet=false     Only display numeric IDs
       -s, --size=false      Display total file sizes
       --since=""            Show created since Id or Name, include non-running
-      --format=[]       Pretty-print containers using a Go template
+      --format=[]           Pretty-print containers using a Go template
+      --help=false          Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/ps.md]: and here

File: docs/reference/commandline/restart.md [hunk 13]
```diff
@@ -16,4 +16,5 @@ weight=1
     Restart a container
 
       -t, --time=10      Seconds to wait for stop before killing the container
+      --help=false       Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/restart.md]: (alphabetical)

File: docs/reference/commandline/rm.md [hunk 14]
```diff
@@ -18,6 +18,7 @@ weight=1
       -f, --force=false      Force the removal of a running container (uses SIGKILL)
       -l, --link=false       Remove the specified link
       -v, --volumes=false    Remove the volumes associated with the container
+      --help=false           Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/rm.md]: (alphabetical)

File: docs/reference/commandline/rmi.md [hunk 15]
```diff
@@ -17,7 +17,7 @@ weight=1
 
       -f, --force=false    Force removal of the image
       --no-prune=false     Do not delete untagged parents
-
+      --help=false         Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/rmi.md]: (alphabetical)

File: docs/reference/commandline/save.md [hunk 16]
```diff
@@ -16,6 +16,7 @@ weight=1
     Save an image(s) to a tar archive (streamed to STDOUT by default)
 
       -o, --output=""    Write to a file, instead of STDOUT
+      --help=false       Print usage
```
Comment (thaJeztah) [inline docs/reference/commandline/save.md]: (alphabetical)

File: man/docker-build.1.md [hunk 17]
```diff
@@ -34,9 +33,9 @@ directory to the Docker daemon. The contents of this directory would
 be used by **ADD** commands found within the Dockerfile.
 
 Warning, this will send a lot of data to the Docker daemon depending
-on the contents of the current directory. The build is run by the Docker
```
Comment (thaJeztah) [inline man/docker-build.1.md]: pls check if this trailing whitespace is intentional (also the line below)

File: man/docker-create.1.md [hunk 18]
```diff
@@ -53,7 +53,7 @@ docker-create - Create a new container
 [**--read-only**[=*false*]]
 [**--restart**[=*RESTART*]]
 [**--security-opt**[=*[]*]]
-[**--stop-signal**[=*SIGNAL*]]
+[**--stop-signal**[=*SIGTERM*]]
```
Comment (thaJeztah) [inline man/docker-create.1.md]: I think the original "SIGNAL" was correct here, these are not the "default" values

**Comment by thaJeztah** [comment]
Thanks @runcom! Added my comments inline; I haven't commented on all *alphabetical* changes, probably easy to find the remaining ones :-)

File: docs/reference/commandline/build.md (hunk 1 above)
Comment (runcom) [inline docs/reference/commandline/build.md]: you know this is a major effort to do, will instead address this in another PR wdyt? at least now we have the help flag :)

File: docs/reference/commandline/cp.md (hunk 3 above)
Comment (runcom) [inline docs/reference/commandline/cp.md]: almost sure 2 spaces are needed here (other *.md fiiles have them)

File: man/docker-create.1.md (hunk 18 above)
Comment (runcom) [inline man/docker-create.1.md]: but -c has 0 for instance, read-only has false, wdyt?

File: docs/reference/commandline/inspect.md (hunk 9 above)
Comment (runcom) [inline docs/reference/commandline/inspect.md]: everything seems fine https://github.com/runcom/docker/blob/fix-mam-commandline/docs/reference/commandline/inspect.md

File: man/docker-build.1.md (hunk 17 above)
Comment (runcom) [inline man/docker-build.1.md]: seems correct :)

File: docs/reference/commandline/inspect.md (hunk 9 above)
Comment (thaJeztah) [inline docs/reference/commandline/inspect.md]: :+1:

**Comment by runcom** [comment]
everything alpha ordered now

**Comment by thaJeztah** [comment]
Oh, wow, thanks! :heart: 

LGTM

ping @moxiegirl @SvenDowideit

**Comment by moxiegirl** [comment]
@runcom This is great.  I'll need to run a `make docs` check on the command outs.  Even though there aren't any significant code changes, this does hit non-.md files so we should have a quick eyeball by a coder too.

**Comment by runcom** [comment]
@moxiegirl fair :)

File: docs/reference/commandline/export.md [hunk 19]
```diff
@@ -12,19 +12,17 @@ weight=1
...
-      -o, --output=""    Write to a file, instead of STDOUT
-
-      Produces a tarred repository to the standard output stream.
+    Export the contents of a container's filesystem as a tar archive
 
+      --help=false       Print usage
+      -o, --output=""    Write to a file, instead of STDOUT
 
- For example:
+## Examples
```
Comment (moxiegirl) [inline docs/reference/commandline/export.md]: The examples section should be after the description:

...on exporting data in a volume.

## Examples

**Comment by vdemeester** [comment]
Code LGTM :wink:

**Comment by moxiegirl** [comment]
@runcom Wonderful job! Thank you for doing this check and correction. I spotted one out of sync Examples section if you have time to fix, that would be nice but really unnecessary considering how nice you were to do *this*.  LGTM.   Once a coder does a quick check, good to merge from a docs perspective.  

@thaJeztah if you want to merge without the code check, that's your call. As courtesy, I don't merge code changes but leave that to coders.

File: docs/reference/commandline/export.md (hunk 19 above)
Comment (runcom) [inline docs/reference/commandline/export.md]: I'll fix this before merge don't worry!

Comment (runcom) [inline docs/reference/commandline/export.md]: fixed

**Comment by runcom** [comment]
ping @thaJeztah as @vdemeester LGTM'ed go files changes :)

**Comment by thaJeztah** [comment]
Heh, ping Janky please merge when you're done!

**Comment by thaJeztah** [comment]
Windows failure is unrelated; https://github.com/docker/docker/issues/16738

**Pull request merged**